controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,221 @@
+	labelSelector := labels.SelectorFromSet(labels.Set{DisruptionScheduleNameLabel: instance.Name})
+
+	if err := r.Client.List(ctx, disruptions, client.InNamespace(instance.Namespace), &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		r.log.Errorw("unable to list Disruptions", "DisruptionSchedule", instance.Name, "err", err)
+		return disruptions, err
+	}
+
+	return disruptions, nil
+}
+
+// updateLastScheduleTime updates the LastScheduleTime in the status of a DisruptionSchedule instance based on the most recent schedule time among the given disruptions
+func (r *DisruptionScheduleReconciler) updateLastScheduleTime(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule, disruptions *chaosv1beta1.DisruptionList) error {
+	mostRecentScheduleTime := r.getMostRecentScheduleTime(disruptions) // find the last run so we can update the status
+	if mostRecentScheduleTime != nil {
+		instance.Status.LastScheduleTime = &metav1.Time{Time: *mostRecentScheduleTime}
+	} else {
+		instance.Status.LastScheduleTime = nil
+	}
+
+	return r.Client.Status().Update(ctx, instance)
+}
+
+// getMostRecentScheduleTime returns the pointer to the most recent scheduled time among a list of disruptions
+func (r *DisruptionScheduleReconciler) getMostRecentScheduleTime(disruptions *chaosv1beta1.DisruptionList) *time.Time {
+	var mostRecentScheduleTime *time.Time
+
+	for _, disruption := range disruptions.Items {
+		scheduledTimeForDisruption, err := r.getScheduledTimeForDisruption(&disruption)
+		if err != nil {
+			r.log.Errorw("unable to parse schedule time for child disruption", "err", err, "disruption", &disruption)
+			continue
+		}
+
+		if scheduledTimeForDisruption != nil {
+			if mostRecentScheduleTime == nil {
+				mostRecentScheduleTime = scheduledTimeForDisruption
+			} else if mostRecentScheduleTime.Before(*scheduledTimeForDisruption) {
+				mostRecentScheduleTime = scheduledTimeForDisruption
+			}
+		}
+	}
+
+	return mostRecentScheduleTime
+}
+
+// getScheduledTimeForDisruption returns the scheduled time for a particular disruption
+func (r *DisruptionScheduleReconciler) getScheduledTimeForDisruption(disruption *chaosv1beta1.Disruption) (*time.Time, error) {
+	timeRaw := disruption.Annotations[ScheduledAtAnnotation]
+	if len(timeRaw) == 0 {
+		return nil, nil
+	}
+
+	timeParsed, err := time.Parse(time.RFC3339, timeRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	return &timeParsed, nil
+}
+
+func (r *DisruptionScheduleReconciler) checkTargetResourceExists(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) (bool, error) {
+	var targetObj client.Object
+
+	switch instance.Spec.TargetResource.Kind {
+	case "Deployment":
+		targetObj = &appsv1.Deployment{}
+	case "StatefulSet":
+		targetObj = &appsv1.StatefulSet{}
+	}
+
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      instance.Spec.TargetResource.Name,
+		Namespace: instance.Namespace,
+	}, targetObj)
+
+	return errors.IsNotFound(err), err
```

Comment:
This is close to being just my personal preference, but I believe it will be easier for callers of this function if the return looks more like:
```
if errors.IsNotFound(err) {
  return false, nil
} else if err != nil {
  return true, nil
} else {
  return false, err
}
```